### PR TITLE
OVM Build Flag

### PR DESCRIPTION
**Pull request by K-Ho**
Adds a `--use-OVM` build flag that will run the build script using the Optimism `solc` fork for compilation. This will be helpful in ensuring contract sizes remain under `24KB` even with the overhead of our compilation process.

### CODE REVIEW

File: publish/src/commands/build.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ const build = async ({
 	testHelpers,
 	showWarnings,
 	showContractSize,
+	useOVM,
```
Comment (justinjmoses) [inline publish/src/commands/build.js:34]: Can you also add a console log in gray to log when the flag is enabled?

Comment (K-Ho) [reply publish/src/commands/build.js:34]: done!

**Comment by justinjmoses** [comment]
Thanks @K-Ho - do you know if buidler supports passing through the solc compiler as well? might be good to have a flag in `buidler.config.js` to allow us to use your ovm-compiler when running tests.

**Comment by K-Ho** [comment]
> Thanks @K-Ho - do you know if buidler supports passing through the solc compiler as well? might be good to have a flag in `buidler.config.js` to allow us to use your ovm-compiler when running tests.

Buidler doesn't support custom solc compilers. To get the tests running in the OVM, we need to import a new dependency and add two builder plugins - one that enables passing a path to a custom solc and another that sets the network provider to be an OVM provider.

**Comment by justinjmoses** [comment]
OK thanks for that. Hopefully a newer version of buidler will make this a lot easier

**Pull request merged**